pkg/testutil/testutil.go

```diff
@@ -17,15 +17,33 @@ package testutil
 import (
 	"net/url"
 	"runtime"
+	"strings"
 	"testing"
+	"time"
 )
 
-// WARNING: This is a hack.
-// Remove this when we are able to block/check the status of the go-routines.
-func ForceGosched() {
-	// possibility enough to sched up to 10 go routines.
-	for i := 0; i < 10000; i++ {
-		runtime.Gosched()
+func runnableGoroutineNumber() int {
```

Comment:
@jonboulle My suggestion is to simply sleep for 5ms for now... There is basically no simple way to do it (or the effort might not worth the outcome). And we have been trying too hard to solve it with some hacks.